src/signalfx_metadata.py

```diff
@@ -883,6 +883,7 @@ def send():
     # day, then once a day from then on but off by a fudge factor
     global NEXT_METADATA_SEND
     if NEXT_METADATA_SEND == 0:
+        set_aws_url(get_aws_info())
```

Comment:
You don't want this here.  That's only going to happen once on startup (in addition to the once in config).

I would put set_aws_url into like 1056 immediately after your AWS=True line.  It has a flag for aws_set to only be done once.  Maybe we change this to only use  one flag that starts as false and goes to true after one successful iteration of becoming aws.  You'll have to think through the logic implications of that, but at worst, just putting it here would solve the problem.